File: Python/2021/1stOct_pradhumna22.py
n = int(input()) 
def printFibo( n):
        
    a = 1
    b = 1
    print(a + " " + b)

    while(n > 2):
        c = a + b
        a = b
        b = c

        if(isPrime(c) or c//5==0):
            print(" " + 0 + " ")
            
        else:
            print(" "+ c + " ")
            
        n=n-1

def isPrime(num):
    for i in range(2, i*i<=num):
        if(num % i == 0):
            return False
        i=i+1
    return True

File: JAVA/2021/1stOct_pradhumna22.java

import java.util.Scanner;

public class Main {
//main method
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        printFibo(n);
    }

//printing function
    public static void printFibo(int n){

        int a = 1, b = 1,c;
        System.out.print(a + " " + b);

        while(n > 2){
            c = a + b;
            a = b;
            b = c;

            if(isPrime(c) || c%5==0)
            {
                System.out.print(" " + 0 + " ");
            }
            else
                {
                System.out.print(" "+ c + " ");
            }

            n--;
        }

    }
//check for prime
    public static boolean isPrime(int num){
        for(int i=2; i*i<=num; i++){
            if(num % i == 0){
                return false;
            }
        }
        return true;
    }


}
